packages/compass-serverstats/src/components/detailview-component.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Button, Icon } from '@mongodb-js/compass-components';
+import { useTelemetry } from '@mongodb-js/compass-telemetry/provider';
+
+import Actions from '../actions';
+
+function removeMS(key: string, value: any) {
+  if (key === 'ms_running') {
+    return undefined;
+  }
+  return value;
+}
+
+type CurrentOpData = {
+  op: string;
+  ns: string;
+  ms_running: number;
+  opid: number;
+  client: string;
+  active: boolean;
+  waitingForLock: boolean;
+};
+
+export function DetailViewComponent() {
+  const [data, setData] = useState<null | CurrentOpData>(null);
+
+  const track = useTelemetry();
+
+  const onShow = useCallback((newDataToShow: CurrentOpData) => {
+    setData(newDataToShow);
+  }, []);
+
+  const onHide = useCallback(() => {
+    setData(null);
+  }, []);
```

Comment:
Small nit: because `.listen` method returns unsubscribe functions, you dont even need extra useCallbacks here, can just define those functions right in the listen methods, less code and runtime